org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -72,9 +83,18 @@ public XtextResource getResourceFor(InputStream stream) {
 		XtextResourceSet resourceSet = resourceSetProvider.get();
 		initializeTypeProvider(resourceSet);
 		try {
-			URI resourceUri = URI.createURI("Test." + fileExtensionProvider.getPrimaryFileExtension());
-			Resource resource = resourceSet.createResource(resourceUri);
-			resource.load(stream, null);
+			String projectFullPath = javaProject.getProject().getFullPath().toString();
+			URI resourceUri = URI.createPlatformResourceURI(projectFullPath + "/" + "Test." + fileExtensionProvider.getPrimaryFileExtension(), true);
+			
+			/*
+			 * Avoid the following java.lang.IllegalStateException:
+			 * A different resource with the URI 'platform:/resource/contentAssistTest/Test....' was already registered.
+			 */
+			Resource resource = resourceSet.getResource(resourceUri, false);
+			if (resource == null) {
```

Comment:
But if the resource is not null, i.e., it exists, the passed input stream is not used to populate it, is it?

Reply:
Yes. Exactly.

Reply:
So I still don't understand... I was expecting the specified input stream to be taken into consideration and in case replace the current resource... maybe I'm still missing something

Reply:
@LorenzoBettini : The method `getResourceFor(InputStream stream)` will be called twice within each test case execution, in both cases the passed stream is created from the `currentModelToParse` variable. So basically it does not matter if we use the first call and ingore the second call or if we create the resource on the first call and overwrite it with the same content in the second call.

I modified the PR so that in the second call the current resource content will be replaced, as requested.

Reply:
@miklossy but the question now is why is it called twice?

Reply:
It is a good question, but currently I am not sure if we could fix that without any breaking change. Let's analyse it in a follow up issue :-)

Reply:
> It is a good question, but currently I am not sure if we could fix that without any breaking change. Let's analyse it in a follow up issue :-)

OK, if it's always been like that :)